docs/development/plugin/development-plugin-resources.asciidoc

```diff
@@ -51,4 +51,24 @@ The Kibana directory must be named `kibana`, and your plugin directory must be l
 If you're developing a plugin that has a user interface, take a look at our https://elastic.github.io/eui[Elastic UI Framework].
 It documents the CSS and React components we use to build Kibana's user interface.
 
-You're welcome to use these components, but be aware that they are rapidly evolving and we might introduce breaking changes that will disrupt your plugin's UI.
\ No newline at end of file
+You're welcome to use these components, but be aware that they are rapidly evolving and we might introduce breaking changes that will disrupt your plugin's UI.
+
+[float]
+==== TypeScript Support
+Plugin code can be written in http://www.typescriptlang.org/[TypeScript] if desired. To enable TypeScript support create a `tsconfig.json` file at the root of your plugin that looks something like this:
+
+["source","js"]
+-----------
+{
+  // extend Kibana's tsconfig, or use your own settings
+  "extends": "../../kibana/tsconfig.json",
+
+  // tell the TypeScript compiler where to find your source files
+  "include": [
+    "server/**/*",
+    "public/**/*"
+  ]
+}
+-----------
+
+TypeScript code is automatically converted into JavaScript during development, but not in the distributable version of Kibana. If you use the {repo}blob/{branch}/packages/kbn-plugin-helpers[@kbn/plugin-helpers] to build your plugin then your `.ts` and `.tsx` files will be permanently transpiled before your plugin is archived. If you have your own build process, make sure to run the TypeScript compiler on your source files and ship the compilation output so that your plugin will work with the distributable version of Kibana.
```

Comment:
sorry, the .jsx extension, not jsx itself

Reply:
I'll let @spalger confirm, but the same question popped into my head when I was reviewing this. The ts docs make it seem like you *must* use `.tsx` for jsx in typescript. I didn't check if the alternative worked because I figured we should probably do what the project recommends regardless.

Reply:
Yeah, it's not optional in TypeScript. JSX support is only available in `.tsx` files.